pkg/plugin/processor/builtin/registry.go

```diff
@@ -185,7 +165,12 @@ func (r *Registry) NewProcessor(_ context.Context, fullName plugin.FullName, _ s
 		return nil, cerrors.Errorf("could not find builtin plugin %q, only found versions %v: %w", fullName, availableVersions, plugin.ErrPluginNotFound)
 	}
 
-	return b.constructor(r.logger), nil
+	p := b.constructor(r.logger)
+
+	// apply default middleware
+	p = sdk.ProcessorWithMiddleware(p, sdk.DefaultProcessorMiddleware(p.MiddlewareOptions()...)...)
```

Comment:
This is the most important part, it's where we apply the middleware to all builtin plugins. The standalone ones get the middleware applied in the processor SDK (see [this line](https://github.com/ConduitIO/conduit-processor-sdk/pull/68/files#diff-05d67ab64d2cc3aeb2a3d4b1b1de5b145c06a2371ddf94660d326f5aa265ab6dR72)).